app/codes/validator.py

```diff
@@ -104,9 +104,10 @@ def get_node_trust_score(public_key):
     return 1
 
 
-def validate_block_receipts(block):
+def count_block_receipts(block):
     total_receipt_count = 0
     positive_receipt_count = 0
+    negative_receipt_count = 0  # TODO
     for receipt in block['receipts']:
```

Comment:
@kousthubraja the receipts being from committee members check can happen here. Ignore the receipts not from committee members. We can use the get_committee_for_current_block.
Refer to comment on line 140 of consensus.py above.

Reply:
it's already verified [here](https://github.com/newrlfoundation/newrl/blob/9aa6e26ad51ffdd8413171b2552074e3070e506c/app/codes/p2p/sync_chain.py#L108)

during block receive, if the block has any receipt not in committee or miner, the entire block will be rejected as no honest node would append a bad receipt to a block.